website/docs/r/compute_route.html.markdown

```diff
@@ -40,52 +71,94 @@ resource "google_compute_route" "default" {
-    route applies to.
+* `dest_range` -
+  (Required)
+  The destination range of outgoing packets that this route applies to.
+  Only IPv4 is supported.
+* `name` -
+  (Required)
+  Name of the resource. Provided by the client when the resource is
+  created. The name must be 1-63 characters long, and comply with
+  RFC1035.  Specifically, the name must be 1-63 characters long and
+  match the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?` which means
+  the first character must be a lowercase letter, and all following
+  characters must be a dash, lowercase letter, or digit, except the
+  last character, which cannot be a dash.
+* `network` -
+  (Required)
+  The network that this route applies to.
 
-* `network` - (Required) The name or self_link of the network to attach this route to.
-
-* `priority` - (Optional) The priority of this route, used to break ties. Defaults to 1000.
 
 - - -
 
-* `next_hop_gateway` - (Optional) The URL of the internet gateway to route
-    to if this route is matched. The alias "default-internet-gateway" can also
-    be used.
+* `description` -
+  (Optional)
+  An optional description of this resource. Provide this property
+  when you create the resource.
+* `priority` -
+  (Optional)
+  The priority of this route. Priority is used to break ties in cases
+  where there is more than one matching route of equal prefix length.
+
+  In the case of two routes with equal prefix length, the one with the
+  lowest-numbered priority value wins.
+
+  Default value is 1000. Valid range is 0 through 65535.
+* `tags` -
+  (Optional)
+  A list of instance tags to which this route applies.
+* `next_hop_gateway` -
+  (Optional)
+  URL to a gateway that should handle matching packets.
+
+  Currently, you can only specify the internet gateway, using a full or
+  partial valid URL:
+
+  * https://www.googleapis.com/compute/v1/projects/project/
+        global/gateways/default-internet-gateway
+  * projects/project/global/gateways/default-internet-gateway
+  * global/gateways/default-internet-gateway
+* `next_hop_instance` -
+  (Optional)
+  URL to an instance that should handle matching packets.
+  You can specify this as a full or partial URL. For example:
+
+  * https://www.googleapis.com/compute/v1/projects/project/zones/zone/
+      instances/instance
+  * projects/project/zones/zone/instances/instance
+  * zones/zone/instances/instance
+* `next_hop_ip` -
+  (Optional)
+  Network IP address of an instance that should handle matching packets.
+* `next_hop_vpn_tunnel` -
+  (Optional)
+  URL to a VpnTunnel that should handle matching packets.
+* `project` (Optional) The ID of the project in which the resource belongs.
+    If it is not provided, the provider project is used.
 
-* `next_hop_instance` - (Optional) The name of the VM instance to route to
-    if this route is matched.
 
 * `next_hop_instance_zone` - (Required when `next_hop_instance` is specified)
```

Comment:
If `next_hop_instance` is specified via URL, I think then this field is no longer required.